src/rubyfmt.rb

```diff
@@ -2401,8 +2209,251 @@ def format_redo(ps, expression)
   ps.emit_newline if ps.start_of_line.last
 end
 
+def format_splat(ps, rest)
+  ps.emit_ident("*")
+  format_expression(ps, rest[0])
+end
+
+def format_to_proc(ps, rest)
+  ps.emit_ident("&")
+  format_expression(ps, rest[0])
+end
+
+def format_keyword(ps, rest)
+  ps.emit_ident(rest[0])
+end
+
+def format_method_call(ps, rest)
+  ps.emit_indent if ps.start_of_line.last
+
+  chain, method, original_used_parens, args = rest
+
+  args = args.map { |a| normalize(a) }
+
+  use_parens = (
```

Comment:
This is pretty awful, but it's as close as I could get to the logic that existed across the various `format_*` methods that are replaced by this method.

Reply:
I have something here.

Reply:
I pulled it out into a method, but further improvements would be great.